src/chunk_insert_state.c

```diff
@@ -22,6 +22,79 @@
 #include "compat.h"
 #include "chunk_index.h"
 
+/* We don't want to create a new constraint on each chunk action as that takes
+ * too much memory. Unfortunately, postgres stores things in expressions with
+ * the assumption that the data will live throughout the expression context. As
+ * a compromise we store a transaction-local cache of the created constraints.
+ */
+static __thread HTAB *constraint_cache = NULL;
+static __thread MemoryContext constraint_cache_context = NULL;
+
+
+#ifdef TIMESCALE_CONSTRAINT_CACHE_STATS
```

Comment:
I don't see this define being set anywhere. If we'd like to keep this as an option, I think it should be a settable cache variable in CMake. Otherwise remove.